pytorch_lightning/loops/base.py

```diff
@@ -335,4 +339,5 @@ def _load_from_state_dict(self, state_dict: Dict, prefix: str, metrics: Optional
                     v.reset(metrics=False)
 
         self.on_load_checkpoint(state_dict[prefix + "state_dict"])
-        self.restarting = True
+        if _fault_tolerant_training():
+            self.restarting = True
```

Comment:
since we are using loop.load_state_dict it's restarting the loop by marking this flag as True. Updating this... IMO load_state_dict shouldn't decide the flag here. It should be set explicitly during the restart.